src/sign.rs

```diff
@@ -0,0 +1,304 @@
+use tiny_keccak::Keccak;
+use secp256k1;
+use secp256k1::key::{SecretKey, PublicKey};
+use secp256k1::{Error, Message, RecoverableSignature, RecoveryId};
+use std::collections::HashMap;
+use num_bigint::{BigUint};
+use rand::{Rng, thread_rng};
+
+lazy_static! {
+	pub static ref SECP256K1: secp256k1::Secp256k1 = secp256k1::Secp256k1::new();
+}
+
+pub trait Encodable {
+    fn encode(&self) -> Vec<u8>;
+}
+
+pub trait ToBytes {
```

Comment:
`Into` can't be implemented for arbitrary types.
How one should handle this, without custom trait?